packages/next-playwright/src/next-worker-fixture.ts

```diff
@@ -0,0 +1,83 @@
+import type { Server } from 'http'
+import type { ProxyRequest } from './proxy-request'
+import { createProxyServer } from './proxy-server'
+
+type MockFetch = (request: Request) => Response | Promise<Response>
+
+export interface NextWorkerFixture {
+  proxyPort: number
+  mockFetch: (testId: string, url: string, mock: MockFetch) => void
+}
+
+class NextWorkerFixtureImpl implements NextWorkerFixture {
+  public proxyPort: number
+  private proxyServer: Server
+  private proxyFetchMap = new Map<string, Map<string, MockFetch>>()
+
+  async setup(): Promise<void> {
+    const server = await createProxyServer({
+      handleRequest: this.handleProxyRequest.bind(this),
+    })
+
+    const address = server.address()
+    if (!address || typeof address !== 'object') {
+      server.close()
+      throw new Error('Failed to create a proxy server')
+    }
+    this.proxyPort = address.port
+    this.proxyServer = server
+  }
+
+  teardown(): void {
+    this.proxyServer.close()
+  }
+
+  mockFetch(testId: string, url: string, mock: MockFetch): void {
+    let map = this.proxyFetchMap.get(testId)
+    if (!map) {
+      map = new Map<string, MockFetch>()
+      this.proxyFetchMap.set(testId, map)
+    }
+    map.set(url, mock)
```

Comment:
There are a few other things we may want/need to support here, which we've built into our internal tools (client-only):
- Multiple mocks per URL.
- Different mocks for different HTTP methods.
- Mock rejections.
- Defaults for endpoints (like a user, where you want to always return the same user for a set of tests).

Reply:
Certainly. This would likely expand significantly. But let's first confirm if this is a good direction.

Reply:
See the MSW prototype in README.md or the implementation in the `msw.ts`.